### ISSUE TITLE
[Truffle] Breaking outer while loop fails

### ISSUE THREAD

**Issue by bjfish** [jruby/jruby#2675]
This causes array#cycle spec to fail

```ruby
def yield_me
  n = 0
  while true
    [0, 1, 2].each do |n|
      puts "hello"
      yield(n)
      puts "shouldn't print this"
    end
  end
end

yield_me { |n| break if n == 0 }
```
Expected output (ruby 2.2, jruby )
```
$ ruby looping.rb 
hello
```
Actual output
```
$ ~/Documents/jruby-mine/bin/jruby -X+T -Xtruffle.printRuntime=true looping.rb
jruby: using Default Truffle Runtime
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
```
... forever

**Comment by chrisseaton** [comment]
This doesn't break out of the while loop - it breaks out of the block lexically enclosing the break - the block on the last line. Same problem as #2826 I think. @eregon is already investigating.

**Comment by eregon** [comment]
Fixed in 434c8a2.
